vms/avm/unique_tx.go

```diff
@@ -163,40 +158,25 @@ func (tx *UniqueTx) Status() choices.Status {
 	return tx.status
 }
 
-// Dependencies returns the set of transactions this transaction builds on
-func (tx *UniqueTx) Dependencies() ([]snowstorm.Tx, error) {
+// MissingDependencies returns the set of transactions that are not currently
+// accepted, but must be accepted before this transaction should be accepted.
+func (tx *UniqueTx) MissingDependencies() (set.Set[ids.ID], error) {
 	tx.refresh()
-	if tx.Tx == nil || len(tx.deps) != 0 {
-		return tx.deps, nil
-	}
-
 	txIDs := set.Set[ids.ID]{}
 	for _, in := range tx.Unsigned.InputUTXOs() {
 		if in.Symbolic() {
 			continue
 		}
 		txID, _ := in.InputSource()
-		if txIDs.Contains(txID) {
-			continue
-		}
-		txIDs.Add(txID)
-		tx.deps = append(tx.deps, &UniqueTx{
+		inputTx := &UniqueTx{
 			vm:   tx.vm,
 			txID: txID,
-		})
-	}
-	consumedIDs := tx.Tx.Unsigned.ConsumedAssetIDs()
-	for assetID := range tx.Tx.Unsigned.AssetIDs() {
-		if consumedIDs.Contains(assetID) || txIDs.Contains(assetID) {
-			continue
 		}
-		txIDs.Add(assetID)
-		tx.deps = append(tx.deps, &UniqueTx{
-			vm:   tx.vm,
-			txID: assetID,
-		})
+		if inputTx.Status() != choices.Accepted {
+			txIDs.Add(txID)
+		}
 	}
```

Comment:
This loop was only needed if someone created a `Fx` with an `Operation` that could be issued with no `Input` `UTXO`s. However, no such `Fx` was ever added - and so it is safe to remove (as all chains are now linearized).